File: 01_10_2025/Q1.java
/*Q1. Write a java program to display top three value from array.
Sample Input: [78, 90, 90, 86, 85, 92, 70, 92]
Expected Output: 92, 90, 86
*/

import java.util.*;
public class Q1{
    public static void main(String x[]) {
	Scanner sc=new Scanner(System.in);

	//input size and aray element form user
	System.out.println("Enter size of array");
	int n=sc.nextInt();
	System.out.println("Enter "+ n +" element in array");
	//declare array
	int arr[]=new int[n];
	for(int i=0;i<n;i++)
	{
	  arr[i]=sc.nextInt();
	}

	//initilized value to variable  for finding
        int first =arr[0];    
        int second=arr[0];  
        int third=arr[0];  


	//finding first sec third largest element
        for (int i=0;i<n;i++)  
	 {
	     int num=arr[i];  
            if (num >first) {
                third=second; 
                second=first; 
                first=num;    

            } else if (num <first && num>second) {   
                third=second;
                second=num;
            } else if(num <second && num>third) { 
                third=num;
            }
        }

	//printing number
        System.out.println(first +", "+ second +", "+third);
    }
}

File: 08_11_2025/UI_Test/js/Q1.js
function findSecondLargest(arr) {
  if (arr.length < 2) {
    console.log("Array does not have enough elements");
    return;
  }

  let first = 0;
  let second = 0;

  for (let i = 0; i < arr.length; i++) {
    let num = arr[i]; //10  20 4 45 99

    if (num === first) continue; // skip same number as first

    if (num > first) {
      second = first;
      first = num;
    } else if (num > second && num < first) {
      //10>0
      second = num;
    }
  }

  if (second === 0) {
    console.log("Array does not have enough distinct elements");
  } else {
    console.log("Second Largest Number:", second);
  }
}

// Example
findSecondLargest([10, 20, 4, 45, 99]); // Output → Second Largest Number: 45
